twilight-http/src/error.rs

```diff
@@ -124,6 +132,7 @@ pub enum ErrorType {
     /// This can occur if a bot token is invalidated or an access token expires
     /// or is revoked. Recreate the client to configure a new token.
     Unauthorized,
+    Validation,
```

Comment:
I think docs are missing here.

I worry a bit that the actual wording of the inner error will be a bit hidden so I think we should add a example here that prints the inner error of a validation error through `source`.